cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/config/ETLBatchConfig.java

```diff
@@ -29,34 +31,49 @@
 public final class ETLBatchConfig extends ETLConfig {
   private final String schedule;
   private final List<ETLStage> actions;
+  private final List<LocalizeResourceInfo> resourcesToLocalize;
```

Comment:
or if the ETL app is smart enough not to localize the same URI multiple times? The main cost of localization involves copying the file to HDFS (if it is not on HDFS) and have NM copy it to local FS. The same HDFS location (timestamp + lenght) only copied once per box, and it will be shared across all containers on that box through symlink.